### ISSUE TITLE
helm test should support Job

### ISSUE THREAD

**Issue by schollii** [helm/helm#9776]
Output of `helm version`:

2.17

Output of `kubectl version`:

1.18

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

AWS with kops

It does not seem possible to define a helm test template to use a Job instead of a plain Pod: 

```
$ helm test RELEASE_NAME
Error: RELEASE_NAME is not a pod
```
The manifest in `tests` folder of chart: 
```
apiVersion: v1
kind: Job
metadata:
  name: {{ .Values.service.name }}-test
  annotations:
    "helm.sh/hook": test-success
spec:
  backoffLimit: 2
  completions: 1
  template:
    spec:
      restartPolicy: OnFailure
      containers:
        - name: {{ .Values.service.name }}-test
          image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
          imagePullPolicy: Always
```

A Job would give more control such as multiple retries, and seems more aligned with the kubernetes way of thinking about these types of tasks (such as running a test).

**Comment by mattfarina** [comment]
Helm v3 should allow you to use jobs. Helm v2 is no longer supported so we won't be backporting this feature. Helm v3 also allows you to have other assets in a test (e.g., configmap or secret).

**Comment by bacongobbler** [comment]
Yep. Helm Hooks can now be written as Jobs as of Helm 3. See https://helm.sh/docs/topics/charts_hooks/#writing-a-hook for more information.

Piling onto what @mattfarina mentioned earlier... Helm 2's support lifecycle ended on November 2020. See [the blog post](https://helm.sh/blog/helm-2-becomes-unsupported/) for more details. As a result, we will no longer be providing further support for Helm 2. Please migrate to Helm 3 for Job support in Helm Hooks (including test hooks).

Closing!

**Comment by schollii** [comment]
Thanks guys good to know

**Comment by SoerenHenning** [comment]
> Helm v3 should allow you to use jobs. Helm v2 is no longer supported so we won't be backporting this feature. Helm v3 also allows you to have other assets in a test (e.g., configmap or secret).

@mattfarina I stumbled upon your comment when trying to find out how to create resources specifically for tests. Unfortunately, there seems to be no documentation on this or am I overlooking something? (Also the [Chart Tests docs](https://helm.sh/docs/topics/chart_tests/) say that tests are created as jobs, while the example shows a test written as a pod. This is probably a mistake, right?)

What I found out is that when annotating a ConfigMap with `"helm.sh/hook": test` and running `helm test <release> --logs`, I'm getting:

```
Error: unable to get pod logs for <configmap-name>: pods <configmap-name> not found
```
